crates/cast/src/starknet_commands/account/deploy.rs

```diff
@@ -30,9 +35,143 @@ pub struct Deploy {
     pub class_hash: Option<String>,
 }
 
+#[allow(clippy::too_many_arguments)]
 pub async fn deploy(
     provider: &JsonRpcClient<HttpTransport>,
-    path: Utf8PathBuf,
+    accounts_file: Utf8PathBuf,
+    name: String,
+    chain_id: FieldElement,
+    max_fee: FieldElement,
+    wait: bool,
+    class_hash: Option<String>,
+    keystore_path: Option<Utf8PathBuf>,
+    account_path: Option<Utf8PathBuf>,
+) -> Result<InvokeResponse> {
+    if let Some(keystore_path_) = keystore_path {
+        let account_path_ = account_path.ok_or_else(|| {
+            anyhow!("--account must be passed and be a path when using --keystore")
+        })?;
+
+        deploy_from_keystore(
+            provider,
+            chain_id,
+            max_fee,
+            wait,
+            keystore_path_,
+            account_path_,
+        )
+        .await
+    } else {
+        if name == String::default() {
+            bail!("No --name value passed")
+        }
+        account_file_exists(&accounts_file)?;
+        deploy_from_accounts_file(
+            provider,
+            accounts_file,
+            name,
+            chain_id,
+            max_fee,
+            wait,
+            class_hash,
+        )
+        .await
+    }
+}
+
+async fn deploy_from_keystore(
+    provider: &JsonRpcClient<HttpTransport>,
+    chain_id: FieldElement,
+    max_fee: FieldElement,
+    wait: bool,
+    keystore_path: Utf8PathBuf,
+    account_path: Utf8PathBuf,
+) -> Result<InvokeResponse> {
+    let contents =
+        std::fs::read_to_string(account_path.clone()).context("Couldn't read account file")?;
+    let mut items: Map<String, serde_json::Value> = serde_json::from_str(&contents)
+        .map_err(|_| anyhow!("Failed to parse account file at {account_path}"))?;
+
+    if items["deployment"].is_null() {
+        bail!("No deployment field in account JSON file");
+    }
+
+    let status = items
+        .get("deployment")
+        .and_then(|deployment| deployment.get("status"))
+        .and_then(serde_json::Value::as_str)
+        .ok_or_else(|| anyhow::anyhow!("Failed to get status from account JSON file"))?;
+
+    if status == "deployed" {
+        bail!("Account already deployed");
+    }
+
+    let salt = FieldElement::from_hex_be(
+        items
+            .get("deployment")
+            .and_then(|deployment| deployment.get("salt"))
+            .and_then(serde_json::Value::as_str)
+            .ok_or_else(|| anyhow::anyhow!("Failed to get salt from account JSON file"))?,
+    )?;
+    let oz_class_hash = FieldElement::from_hex_be(
+        items
+            .get("deployment")
```

Comment:
it seems to me we could just have a `deployment` variable that we can take various things from, instead of doing .get("deployment") multiple times. this is just a nitpick though, I leave the decision on changing this up to you.